services/util.py

```diff
@@ -96,39 +98,69 @@ def apollo(name, payload):
     return r.json()
 
 
-def parse_adaptor_string(adaptor_input: str) -> tuple[str, str, str]:
+def get_db_connection():
+    """Get database connection from POSTGRES_URL environment variable.
+
+    Returns:
+        psycopg2.connection or None: Database connection if POSTGRES_URL is set, None otherwise
     """
-    Parse adaptor string in format "@openfn/language-http@3.1.11" and return (adaptor_name, version, full_string).
+    db_url = os.environ.get("POSTGRES_URL")
+    if db_url:
+        return psycopg2.connect(db_url)
+    return None
 
-    Accepts:
-    - "@openfn/language-http@3.1.11" -> ("@openfn/language-http", "3.1.11", "@openfn/language-http@3.1.11")
-    - "http@3.1.11" -> ("@openfn/language-http", "3.1.11", "@openfn/language-http@3.1.11")
 
-    Raises ApolloError if version is not provided.
+class AdaptorSpecifier:
```

Comment:
That's lovely - I think that will make handling adaptor strings a lot cleaner :ok_hand: